### ISSUE TITLE
Cannot double interfaces that declare a constructor with PHP 8

### ISSUE THREAD

**Issue by MaxSem** [sebastianbergmann/phpunit#4139]
| Q                   | A
| --------------------| ---------------
| PHPUnit version     | 9.0.1
| PHP version         | 8.0.0-dev
| Installation Method | Composer

#### Summary
```
$ cat test.php
<?php

interface ITest {
        function __construct();
}

class Test extends PHPUnit\Framework\TestCase {
                public function testMock() {
                                $mock = $this->createMock(ITest::class);
                }

                public function testMockBuilder() {
                                $mock = $this->getMockBuilder(ITest::class)->getMock();
                }
}

$ php --version
PHP 8.0.0-dev (cli) (built: Mar 17 2020 16:36:19) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.0-dev, Copyright (c) Zend Technologies

$ php vendor/bin/phpunit test.php
PHPUnit 9.0.1 by Sebastian Bergmann and contributors.


Fatal error: Class Mock_ITest_d9ac1e16 contains 1 abstract method and must therefore be declared abstract or implement the remaining methods (ITest::__construct) in .../vendor/phpunit/phpunit/src/Framework/MockObject/MockClass.php(42) : eval()'d code on line 3
```

#### Expected behavior

For example, older PHPUnit on PHP 7.2:
```
PHPUnit 8.5.2 by Sebastian Bergmann and contributors.

RR                                                                  2 / 2 (100%)

Time: 187 ms, Memory: 6.00 MB

There were 2 risky tests:

1) Test::testMock
This test did not perform any assertions

.../test.php:8

2) Test::testMockBuilder
This test did not perform any assertions

.../test.php:12

OK, but incomplete, skipped, or risky tests!
Tests: 2, Assertions: 0, Risky: 2.
```

[An example of such interface in the wild](https://github.com/wikimedia/mediawiki/blob/master/includes/libs/rdbms/loadbalancer/ILoadBalancer.php)

I know PHP 8 is still in the early stages of development, however PHPUnit claims to support it, so I'd rather make this issue known.

**Comment by sebastianbergmann** [comment]
While PHPUnit is tested on PHP 8 in continuous integration, PHPUnit is not supported on PHP 8 yet.

**Comment by sebastianbergmann** [comment]
#### Code generated with PHP 7.4
```php
class Mock_I_abd6e555 implements PHPUnit\Framework\MockObject\MockObject, I
{
    use \PHPUnit\Framework\MockObject\Api;
    use \PHPUnit\Framework\MockObject\Method;
    use \PHPUnit\Framework\MockObject\MockedCloneMethod;

    public function __construct()
    {
        $__phpunit_arguments = [];
        $__phpunit_count     = func_num_args();

        if ($__phpunit_count > 0) {
            $__phpunit_arguments_tmp = func_get_args();

            for ($__phpunit_i = 0; $__phpunit_i < $__phpunit_count; $__phpunit_i++) {
                $__phpunit_arguments[] = $__phpunit_arguments_tmp[$__phpunit_i];
            }
        }

        $__phpunit_result = $this->__phpunit_getInvocationHandler()->invoke(
            new \PHPUnit\Framework\MockObject\Invocation(
                'I', '__construct', $__phpunit_arguments, '', $this, false
            )
        );

        return $__phpunit_result;
    }
}
```
#### Code generated with PHP 8.0
```php
class Mock_I_1b966968 implements PHPUnit\Framework\MockObject\MockObject, I
{
    use \PHPUnit\Framework\MockObject\Api;
    use \PHPUnit\Framework\MockObject\Method;
    use \PHPUnit\Framework\MockObject\MockedCloneMethod;
}
```
**TL;DR** The constructor is missing in the code generated with PHP 8.

**Comment by sebastianbergmann** [comment]
The root cause for this is that `ReflectionMethod::isConstructor()` behaves differently in PHP 7 and PHP 8.

Details can be found [here](https://news-web.php.net/php.internals/109377).
